src/main/java/com/breskul/bibernate/persistence/GenericDao.java

```diff
@@ -59,28 +60,32 @@ public class GenericDao {
   private static final Logger log = LoggerFactory.getLogger(GenericDao.class);
 
   // TODO: change to select '*'
-  private static final String SELECT_BY_FIELD_VALUE_QUERY = "SELECT %s FROM %s WHERE %s = ?";
+  private static final String SELECT_BY_FIELD_VALUE_QUERY = "SELECT %s FROM %s WHERE %s = ? %s";
   private static final String UPDATE_SQL = "UPDATE %s SET %s WHERE %s = ?;";
   private static final String INSERT_ENTITY_QUERY = "INSERT INTO %s (%s) VALUES (%s);";
   private static final String DELETE_ENTITY_QUERY = "DELETE FROM %s WHERE %s = ?;";
 
   private final Connection connection;
   private final PersistenceContext context;
 
-  public GenericDao(Connection connection, PersistenceContext context) {
+  private final Dialect dialect;
+
+  public GenericDao(Connection connection, PersistenceContext context, Dialect dialect) {
     this.connection = connection;
     this.context = context;
+    this.dialect = dialect;
   }
 
   /**
-   * Find by primary key. Search for an entity of the specified class and primary key. If the entity
-   * instance is contained in the persistence context, it is returned from there.
+   * Find by primary key. Search for an entity of the specified class and primary key. If the entity instance is contained in the
+   * persistence context, it is returned from there.
    *
-   * @param cls – entity class
-   * @param id  - primary key
+   * @param cls      – entity class
+   * @param id       - primary key
+   * @param lockType - the lock mode with which we do select
    * @return the found entity instance or null if the entity does not exist
    */
-  public <T> T findById(Class<T> cls, Object id) {
+  public <T> T findById(Class<T> cls, Object id, LockType lockType) {
```

Comment:
`lockType` is not used in this method at all.

Reply:
Thanks, looks like mincorect changes applied while merging